contracts/zero-ex/contracts/src/features/UniswapFeature.sol

```diff
@@ -167,17 +167,24 @@ contract UniswapFeature is
 
                             let rdsize := returndatasize()
 
-                            returndatacopy(0xB00, 0, rdsize) // reuse memory
-
-                            // Check for ERC20 success. ERC20 tokens should return a boolean,
-                            // but some don't. We accept 0-length return data as success.
+                            // Copy only the first 32 bytes of return data. We
+                            // only care about reading a boolean in the success
+                            // case, and we discard the return data in the
+                            // failure case.
+                            returndatacopy(0xC00, 0, 0x20)
```

Comment:
Simbot is reverting on WETH->DAI when no allowance is set on the EP. I *think* this is because the WETH contract reverts with no data so this `returndatacopy()` tries to read beyond the end of calldata?

Reply:
I do think it's worth clamping the `returndatacopy()` to `min(0x20, rdsize)` in case we ever start reserving the `0xD00+` block in a future update or something.

Or does something like this work?
```solidity
mstore(0xC00, 0)
call(... 0xC00, 0x20) // Write 32 bytes of result to 0xC00
```
Or will it also revert if the call doesn't return at least 32 bytes?

Reply:
Ah. I'll test it out.

Reply:
^Just tried it and it seems to work.

Reply:
That works. But the `mstore(0xC00, 0)` is unnecessary. We don't read from there unless the return data size is at least 32 bytes.

Reply:
Okay, wait, I like your idea better. Write a 0, maybe overwrite with returned data from the call, and then just read the data. Changing code again.

Reply:
Er, wait, that doesn't work. In the 0-bytes-of-return-data case, we need to consider that success. So I think the current code is good.